src/states/Midroll.js

```diff
@@ -0,0 +1,52 @@
+import videojs from 'video.js';
+
+import {AdState} from '../states.js';
+import {startAdBreak, endAdBreak} from '../adBreak.js';
+
+export default class Midroll extends AdState {
+
+  init(player) {
+    this.start();
+  }
+
+  /*
+   * Midroll breaks happen when the integration calls startLinearAdMode,
+   * which can happen at any time during content playback.
+   */
+  start() {
+    const player = this.player;
+
+    if (!this.inAdBreak() && !this.isContentResuming()) {
+      player.ads.debug('startLinearAdMode (Midroll)');
+      player.ads.adType = 'midroll';
+      startAdBreak(player);
+    } else {
+      videojs.log.warn('Unexpected startLinearAdMode invocation (Midroll)');
+    }
+  }
+
+  /*
+   * Midroll break is done.
+   */
+  endLinearAdMode() {
+    const player = this.player;
+
+    if (this.inAdBreak()) {
+      endAdBreak(player);
+      this.contentResuming = true;
+    }
+  }
+
+  /*
+   * End midroll break if there is an error.
+   */
+  onAdsError(player) {
```

Comment:
I believe we should have an `adskip` handler for midrolls too?

Reply:
I don't think so, adskip has to happen before the ad break starts, and in midrolls the ad break starts right away.